fix basic callbacks to work with splitwindow

pik opened the pull request:
This fixes callbacks for: cut, paste, copy, delete, select-all to function correctly with the built-in splitwindow plugin. https://github.com/geany/geany/issues/460

b4n commented on src/callbacks.c:
```diff
@@ -307,16 +307,15 @@ G_MODULE_EXPORT void on_redo1_activate(GtkMenuItem *menuitem, gpointer user_data
 	GtkWidget *focusw = gtk_window_get_focus(GTK_WINDOW(main_widgets.window));
 
-	if (GTK_IS_EDITABLE(focusw))
+	if (GTK_IS_EDITABLE(focusw)) {
```
please follow our coding style (see HACKING, or simply the code around)

pik replied:
Sorry about that, there was some ambiguity i.e. 
https://github.com/geany/geany/blob/master/src/callbacks.c#L279 vs https://github.com/geany/geany/blob/master/src/callbacks.c#L235. Do you prefer single-line if statements to be non-nested in general?

b4n replied:
we don't have a strict rule on whether to have brackets or not on single statements blocks, but we definitely put the brackets on their own lines :)

And regarding your references, don't reproduce the `else if` with each keyword on separate lines, I don't know why it's like it there but we don't want that to spread (e.g. that part in what you changed is good).

pik replied:
> we don't have a strict rule on whether to have brackets or not on single statements blocks, but we definitely put the brackets on their own lines :)

Fixed, thanks. ( It's not my favorite habit but it's the version which doesn't break the go compiler =/  )

b4n commented on src/keybindings.c:
```diff
@@ -1827,11 +1827,11 @@ static void goto_matching_brace(GeanyDocument *doc)
+	GtkWidget *focusw = gtk_window_get_focus(GTK_WINDOW(main_widgets.window));
 
-	if (doc == NULL)
+	if (!IS_SCINTILLA(focusw)) {
```
I didn't test, but that seem to be a problem, as `on_cut1_activate()` and friends are supposed to also work on non-Scintilla text widgets (to allow our shortcut to work everywhere).

But the idea of using the focus widget for `COPYLINE` and `CUTLINE` seem sensible.

pik replied:
@b4n They don't work for me in places I would like currently (i.e. the file viewer pane). The previous code has the same properties (except it also returned if focusw targeted a splitwindow because that would have a different sci object from doc->editor->sci). 
```if (doc == NULL || focusw != GTK_WIDGET(doc->editor->sci)) ```

b4n replied:
Works for me, e.g. in the toolbar search entry, or the scribble pane (just tested with paste, after setting it to a non-default key to be sure it wasn't the widget itself handling it)

pik replied:
@b4n I believe this is because the signal for the find or search and replace dialog is not captured there. The file-browser pane is on the other hand (I assume because it gets focus for the main_window) and fails.

b4n replied:
> @b4n I believe this is because the signal for the find or search and replace dialog is not captured there.

No the search *dialog* is no currently handled, only things in the main window are.  But I was talking about the *toolbar* search entry, which is in the main window ;)

And I'm talking about `cb_func_clipboard_action()`, which does work, not `cb_func_select_action()` which indeed didn't (but for `SELECT_ALL`).

pik replied:
@b4n Ah, I see what you're saying now. It should support all the options which the relevant callbacks support: 
e.g.
```
	if (GTK_IS_EDITABLE(focusw))
		gtk_editable_cut_clipboard(GTK_EDITABLE(focusw));
	else if (IS_SCINTILLA(focusw))
		sci_cut(SCINTILLA(focusw));
	else if (GTK_IS_TEXT_VIEW(focusw))
```
In which case I think it might make sense to just remove the check altogether and let the individual callbacks determine what is and isn't a valid target, since they do that anyways.

b4n commented on src/callbacks.c:
```diff
@@ -1279,10 +1271,10 @@ G_MODULE_EXPORT void on_print1_activate(GtkMenuItem *menuitem, gpointer user_dat
+	g_return_if_fail(IS_SCINTILLA(focusw));
 
-	sci_select_all(doc->editor->sci);
+	sci_select_all(SCINTILLA(focusw));
```
Hum, this callback is odd, and probably should rather call `GEANY_KEYS_SELECT_ALL` to support everything… or maybe the other way around, for this to support everything and be called by `cb_func_select_action()`.

Though, this is a mostly separate issue.

b4n commented on src/keybindings.c:
```diff
@@ -2389,18 +2388,18 @@ static gboolean cb_func_select_action(guint key_id)
 		return TRUE;
 	}
+    else if ((key_id == GEANY_KEYS_SELECT_ALL) && IS_SCINTILLA(focusw)) {
+        on_menu_select_all1_activate(NULL, NULL);
```
it should probably `return` like the others

b4n commented on src/keybindings.c:
```diff
@@ -2411,10 +2410,10 @@ static gboolean cb_func_select_action(guint key_id)
 		case GEANY_KEYS_SELECT_WORDPARTRIGHT:
-			sci_send_command(sci, SCI_WORDPARTRIGHTEXTEND);
+			sci_send_command(doc->editor->sci, SCI_WORDPARTRIGHTEXTEND);
 			break;
```
those two (`WORDPARTLEFT` and `WORDPARTRIGHT`) could also work with the `focusw`

b4n commented:
The general idea seem reasonable to me indeed

pik commented:
Updated: 
* fixed copy/cut/paste regression for toolbar
* copy/cut/paste/select_all now works for file_browser 
* menu select_all fixed to have same behaviour as keybind 

If this looks good `on_menu_select_all1_activate` should probably be renamed to something more generic.

b4n commented on src/keybindings.c:
```diff
@@ -1848,10 +1845,14 @@ static gboolean cb_func_clipboard_action(guint key_id)
-			sci_send_command(doc->editor->sci, SCI_LINECOPY);
+			if (!IS_SCINTILLA(focusw)) 
+				break;
+			sci_send_command(SCINTILLA(focusw), SCI_LINECOPY);
```
style: I'd prefer to see `if (IS_SCINTILLA(...)) sci_send_command()` than `if (! IS_SCINTILLA(...)) break; sci_send_command()`.

pik commented:
Update2: 
Fixed style per b4n's comments.

pik commented:
I'm not sure what you mean, I only removed the `doc != NULL` check since get_document is no longer being used? (I think it's the same in the old diff: https://github.com/pik/geany/commit/63626a4d20bcbd924aeb9ed956c3de31b167e2a1)

b4n commented:
no: https://github.com/pik/geany/commit/63626a4d20bcbd924aeb9ed956c3de31b167e2a1#diff-19f08fc67d9bc13b9de703aa74e97504R2402 vs. https://github.com/pik/geany/commit/7fe1f54fec537c57e585bb67d32e775ba93c8e15#diff-19f08fc67d9bc13b9de703aa74e97504R2402

b4n commented:
this doesn't really matter tough as the next commit removes it anyway

pik commented:
Oh I see, I edited that commit so that the change to `SCINTILLA(focusw)` would be in the select_all reorg, I guess to be more consistent I should put all select changes in the second commit.

b4n commented:
Apart those two small details, LGTM.
BTW, if you want I can very well fix those two small issues myself amending your commits.

pik commented:
> This is not correct, this part was already fixed in the previous commit ;)

I moved all the changes to `cb_func_select_action` the second commit for historical purposes.

b4n commented:
yes, but not [the initial change to `on_menu_select_all1_activate()`](https://github.com/pik/geany/commit/371a4d061bc63c3ee63c3de1f4e01b9385e545b4#diff-4ff3c034cd27f0d7ee45e073f2cb86c1R1187).  But as said don't worry I can do that small detail very easily.

b4n commented:
ah, wait, my bad.  Indeed, while the first commit makes `menu_select_all` work with the splictindow, only the second one actually fixes `select_all`.

b4n commented:
Merged as 2110fc785d77355600229b9d41c399c565d13bf9 (with reflowed 371a4d0) thanks!
